ui/ts/models/status.ts

```diff
@@ -110,133 +58,78 @@ module Models {
         };
-
-        let data: StoreStatusMap = this._data.result();
-        let keys: string[] = Object.keys(data);
-        for (let i: number = 0; i < keys.length; i++) {
-          let storeStatus: Proto.StoreStatus = data[keys[i]];
+        _.each(list, (storeStatus: Proto.StoreStatus) => {
           Proto.AccumulateStatus(status, storeStatus);
-        }
-
-        return m("div", [
-          m("h2", "Details"),
-          m("table", [
-            m("tr", [m("td", "Updated at:"), m("td", _formatDate(status.updated_at))]),
-            m("tr", [m("td", "Ranges:"), m("td", status.range_count)]),
-            m("tr", [m("td", "Leader Ranges:"), m("td", status.leader_range_count)]),
-            m("tr", [m("td", "Available Ranges:"), m("td", status.available_range_count)]),
-            m("tr", [m("td", "Availablility:"), m("td", _availability(status))]),
-            m("tr", [m("td", "Under-Replicated Ranges:"), m("td", status.leader_range_count - status.replicated_range_count)]),
-            m("tr", [m("td", "Fully Replicated:"), m("td", _replicated(status))])
-          ]),
-          Stats.CreateStatsTable(status.stats)
-        ]);
+        });
+        return status;
+      });
+
+      constructor() {
+        this.allStatuses = this._data.result;
       }
-    }
 
-    export interface NodeStatusResponseSet {
-      d: Proto.NodeStatus[];
-    }
+      public GetTotalStatus(): Proto.Status {
+        return this._totalStatus();
+      }
 
-    export interface NodeStatusMap {
-      [nodeId: string]: Proto.NodeStatus;
+      public GetStatus(storeId: string): Proto.StoreStatus {
+        return this._dataMap()[storeId];
+      }
+
+      public refresh(): void {
+        this._data.refresh();
+      }
     }
 
     export class Nodes {
       public allStatuses: Utils.ReadOnlyProperty<Proto.NodeStatus[]>;
 
-      private _data: Utils.QueryCache<NodeStatusMap> = new Utils.QueryCache((): promise<NodeStatusMap> => {
+      private _data: Utils.QueryCache<Proto.NodeStatus[]> = new Utils.QueryCache((): promise<Proto.NodeStatus[]> => {
         return Utils.Http.Get("/_status/nodes/")
           .then((results: NodeStatusResponseSet) => {
-            let data: NodeStatusMap = {};
-            results.d.forEach((status: Proto.NodeStatus) => {
-              let nodeId: number = status.desc.node_id;
-              data[nodeId] = status;
-            });
-            return data;
+            return results.d;
           });
       });
 
-      constructor() {
-        this.allStatuses = Utils.Computed(this._data.result, (map: NodeStatusMap) => {
-          let keys = Object.keys(map).sort();
-          return keys.map((key: string) => map[key]);
+      private _dataMap: Utils.ReadOnlyProperty<NodeStatusMap> = Utils.Computed(this._data.result, (list: Proto.NodeStatus[]) => {
+        let map: NodeStatusMap = {};
+        _.each(list, (status: Models.Proto.NodeStatus) => {
+          let nodeId: number = status.desc.node_id;
```

Comment:
I replaced both of these with _.indexBy(), which is lodash's built in method for doing this (moving lists to a map).